src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StructuralVariationDiscoveryPipelineSpark.java

```diff
@@ -91,17 +111,45 @@ protected void runTool( final JavaSparkContext ctx ) {
         final List<EvidenceTargetLink> evidenceTargetLinks = assembledEvidenceResults.getEvidenceTargetLinks();
         final PairedStrandedIntervalTree<EvidenceTargetLink> evidenceLinkTree = makeEvidenceLinkTree(evidenceTargetLinks);
 
+        final SVIntervalTree<VariantContext> cnvCalls;
+        if (discoverStageArgs.cnvCallsFile != null) {
+            cnvCalls = DiscoverVariantsFromContigAlignmentsSAMSpark.loadCNVCalls(discoverStageArgs.cnvCallsFile, header, sampleId);
+        } else {
+            cnvCalls = null;
+        }
+        
+        final Broadcast<SVIntervalTree<VariantContext>> broadcastCNVCalls;
+        if (cnvCalls != null) {
+            broadcastCNVCalls = ctx.broadcast(cnvCalls);
+        } else {
+            broadcastCNVCalls = null;
+        }
+
         // discover variants and write to vcf
         DiscoverVariantsFromContigAlignmentsSAMSpark
                 .discoverVariantsAndWriteVCF(
                         parsedAlignments,
-                        evidenceLinkTree, assembledEvidenceResults.getReadMetadata(), discoverStageArgs,
+                        discoverStageArgs,
                         ctx.broadcast(getReference()),
-                        header.getSequenceDictionary(), vcfOutputFileName,
-                        localLogger
+                        ctx.broadcast(header.getSequenceDictionary()),
+                        vcfOutputFileName,
+                        localLogger,
+                        evidenceLinkTree,
+                        assembledEvidenceResults.getReadMetadata(),
+                        broadcastCNVCalls
                 );
     }
 
+    public static String getSampleId(final SAMFileHeader header) {
```

Comment:
I think this method is better living in a "utils" class, like `SVUtils`?

Reply:
Moved to `SVUtils`